### PR TITLE
SCHED-333: Data Improvements.

### PR DESCRIPTION
**Pull request by sraaphorst**
First attempt at some improvements in model for data accessibility.

### CODE REVIEW

File: scheduler/core/components/optimizer/dummy.py [hunk 1]
```diff
@@ -42,7 +44,7 @@ def _run(self, plans: Plans):
         while not plans.all_done() and len(self.groups) > 0:
 
             ran_group = random.choice(self.groups)
-            if self.add(ran_group, plans):
+            if self.add(ran_group, plans, None):
```
Comment (stroncod) [inline scheduler/core/components/optimizer/dummy.py:47]: Better do:
```def add(self, group: GroupData, plans: Plans, interval: Optional[Interval] = None):```
so we don't need to pass None as a parameter.

File: scheduler/core/plans/__init__.py [hunk 2]
```diff
@@ -69,6 +68,10 @@ def __init__(self, night_events: Mapping[Site, NightEvents], night_idx: int):
                                         site,
                                         len(ne.times[night_idx]))
 
+    @property
+    def time_slot_length(self):
+        return 0
```
Comment (stroncod) [inline scheduler/core/plans/__init__.py:73]: Why return 0 here?

**Review by stroncod — APPROVED** [review]

File: scheduler/core/components/optimizer/dummy.py (hunk 1 above)
Comment (sraaphorst) [reply scheduler/core/components/optimizer/dummy.py:47]: I was going back and forth trying to decide. Either way works for me, so I'll give it the default parameter.

File: scheduler/core/plans/__init__.py (hunk 2 above)
Comment (sraaphorst) [reply scheduler/core/plans/__init__.py:73]: Ooops. I have no idea why I did this. *doh* I'll fix this.

**Comment by sraaphorst** [comment]
@stroncod I redid this PR because for some reason, it was breaking a test case and I couldn't rebase and get it to work. I went ahead and merged it and I'm closing this one.

**Pull request closed without merging**